hvplot/converter.py

```diff
@@ -1363,6 +1362,10 @@ def method_wrapper(ds, x, y):
             if self._dim_ranges.get('c', (None, None)) != (None, None):
                 style['clim'] = self._dim_ranges['c']
 
+        if self.geo:
+            import geoviews as gv
+            obj = gv.project(obj, projection=self.proj_crs)
```

Comment:
Wondering if this should be after processed = operation() because I worry this eagerly loads the entire dataset.

Reply:
It is needed before the operation. If you move it after it, you get the following figure. We could add more checks to the if statement so we only do this when it is absolutely needed. 

![image](https://github.com/holoviz/hvplot/assets/19758978/18d140b1-7fba-4d21-b758-7c01372876d2)

``` python
import hvplot.xarray
import xarray as xr
import cartopy.crs as ccrs

ds = xr.tutorial.open_dataset("air_temperature")
ds.hvplot.points(
    "lon",
    "lat",
    crs=ccrs.PlateCarree(),
    features=["coastline", "land"],
    xlim=(-140, -80),
    ylim=(20, 45),
    projection=ccrs.LambertConformal(),
    rasterize=True,
    dynspread=True,
)
```

Reply:
Yes I support more checks. I imagine someone has a huge dataset, and gv.project causes it to eagerly load, causing their CPU to crash. Not sure if this should be implemented in another PR, or what the best way forward is. 

Without this change, datashade/rasterize is effectively broken with projections?

Reply:
I have added one more check. I don't think a HoloViews operation will load data more eagerly than another. Both `gv.project` and `rasterize` are operations, which should work with different data formats. 

By the look of it, it does not work great.